src/System.Private.CoreLib/src/System/Threading/Overlapped.cs

```diff
@@ -242,23 +181,22 @@ internal IntPtr UserHandle
 
     public class Overlapped
     {
-        private OverlappedData m_overlappedData;
-        private static PinnableBufferCache s_overlappedDataCache = new PinnableBufferCache("System.Threading.OverlappedData", () => new OverlappedData());
+        private OverlappedData _overlappedData;
 
         public Overlapped()
         {
-            m_overlappedData = (OverlappedData)s_overlappedDataCache.Allocate();
-            m_overlappedData.m_overlapped = this;
+            // The split between Overlapped and OverlappedData should not be needed. It is required by the implementation of 
+            // async GC handles currently. It expects OverlappedData to be a sealed type.
+            _overlappedData = new OverlappedData();
+            _overlappedData._overlapped = this;
```

Comment:
@jkotas, I'm annotating this for nullability... any reason I can't make OverlappedData's ctor take the Overlapped as an argument?  It would allow for the _overlapped field to then be readonly rather than nullable.

Reply:
> OverlappedData's ctor take the Overlapped as an argument

This should be fine. I do not see any issues with it.

Reply:
Thanks.